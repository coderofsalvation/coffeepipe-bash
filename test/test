#!/bin/bash 

{
  cd "$(dirname "$(readlink -f "$0")" )/.."

  cfp(){ cat - | ./cli.coffee "$*" ; }
      
  echo "one,two,three" | cfp '(s) -> s.split ","'

  # detect json / indent json
  echo "[1,2,3,4]"  | cfp '(s) -> JSON.stringify(s,null,2)'

  # search / replace
  echo "foo"        | cfp '(s) -> s.replace /foo/, "bar"'

  # process line by line
  echo -e "foo\nbar" | cfp '(s) -> ( "output: #{line}" for line in s.split "\n")'

  # filter by coffeescript file
  echo -e "foo\nbar" | cfp ./test/filter.coffee 
} | tee /tmp/out

diff -Nau test/test.out /tmp/out && echo OK || { echo ERROR && exit 1; }
