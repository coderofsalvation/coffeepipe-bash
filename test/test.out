one
two
three
[
  1,
  2,
  3,
  4
]
bar
output: foo
output: bar
output: foo
output: bar
